api/apps/v1alpha1/application_defaults.go

```diff
@@ -255,6 +273,39 @@ func defaultObjectiveName(values ...string) string {
 	return defaultName
 }
 
+func defaultCustomObjectiveName(custom *CustomObjective) string {
+	var query string
+	switch {
+	case custom.Prometheus != nil:
+		query = custom.Prometheus.Query
+	case custom.Datadog != nil:
+		query = custom.Datadog.Query
+	}
+
+	// This just tries to make a name out of the query by stripping out
+	// the non-alphanumerics and some token words.
+	var name []string
+	for _, part := range strings.Split(strings.Map(func(r rune) rune {
+		r = unicode.ToLower(r)
+		if (r >= 'a' && r <= 'z') || (r >= '0' && r <= '9') {
+			return r
+		}
+		return '-'
+	}, query), "-") {
+		switch part {
+		case "", "or", "and", "unless", "on", "ignoring", "scalar":
+			continue
+		default:
+			name = append(name, part)
+		}
+	}
+	if n := strings.Join(name, "-"); n != "" {
+		return n
+	}
```

Comment:
Oof. 

Using the queries listed in https://prometheus.io/docs/prometheus/latest/querying/examples/ we get -- 
https://play.golang.org/p/6yTYOB7tujz

I don't think I have a better solution unfortunately.
I thought about doing a sha/checksum of the query and using that instead, but because these are metrics it'll make their usefulness pretty much 0.  Is it worth doing that to nudge people into making pretty names for these metrics but not prohibitive for getting started?